services/service-core/src/auth/keycloak.service.ts

```diff
@@ -81,18 +88,11 @@ export class KeycloakService {
 
   async editUser(id: string, userInput: UpdateUser) {
     const accessToken = await this.getAdminToken();
-    if (
-      userInput.firstName?.trim() === '' ||
-      userInput.lastName?.trim() === '' ||
-      userInput.username?.trim() === ''
-    ) {
+    const { error } = userInputValidator.validate(userInput);
```

Comment:
`error` object will have a lot of information about what exactly is wrong with the input. Consider improving the error message and adding more relevant information